### DISCUSSION TITLE
Area zonal stats - mode for simple copy of attributes

### DISCUSSION THREAD

**Discussion by alronlam** [thinkingmachines/geowrangler#121] [Enhancements Requests]
Hi!

**Context**
I'm just testing out the scenario where the data is not something that needs to be distributed/dis-aggregated.

For example, see screenshot below: 

- The big tile is an Ookla data tile (~650m x 650m), while the smaller tiles are the gridded-AOI tiles.
- The Ookla tile has a mean download speed of 31,854 kbps.

![ookla_data](https://user-images.githubusercontent.com/1049495/183931880-8135f0a2-a1cc-4a24-adee-92c78ace67fb.png)
 
- For the highlighted AOI tile, it was assigned a mean download speed of 13,984 kbps. Likely due to the partial intersection. 
![area_data](https://user-images.githubusercontent.com/1049495/183932967-e601cbfa-a082-4778-8c3e-b6d09ec1a632.png)

**Ask**
However, this might not always be desirable, as the download speed is not something you need to "distribute" (unlike, say, population). 

In this edge case, I'd probably just want the data tile to inherit the attribute of the tile it intersected the most with.

To address these kinds of scenarios, I think we can either (or do both):

1.  Give a param that configures a mode where you'd just copy over attributes of the data tile an AOI tile intersected the most with; or
2. Allow for a mode that performs a weighted average across all intersected tiles, but do NOT include non-intersections. That is, it will still get 100% of the intersecting data tile because it was the only one that intersected. 

Another example to further illustrate Options 1 and 2, in the ff scenario:
![3_intersections](https://user-images.githubusercontent.com/1049495/183938718-ff521895-6b72-476b-a383-c85567623913.png)

1. If we do Option 1, it would copy the attributes of the lower-left data tile (the big blue tile).
2. If we do Option 2, it would get weighted average of the speeds of the three intersecting tiles. For example, get 50% of the speed of the lower-left data tile, 30% of the upper-right data tile, and 20% of the upper-left data tile. That is, the non-intersecting portion of the AOI tile will NOT drag the speed down due to it being "0".

**Comment by butchtm** [comment]
hi @alronlam - there is actually a somewhat undocumented (and still experimental)  feature to allow you to correctly distribute statistics (ie. Option 2)  like "download speed" which you mention above (to disregard the non-intersecting part of the highlighted aoi) so that it only uses the data tiles which it does overlap (and it does the proportionate weighing (based on the overlap on the aoi) of the statistic in proportion to the means of the overlapping data tiles) -- this is by modifying the `func` argument to add an "imputed_" prefix to the statistic if you want to disregard the area not intersected by any data tiles.
e.g. 
```
aoi = azs.create_area_zonal_stats( 
           aoi, 
           data,
           aggregations=[
                      dict(
                         func='imputed_mean',
                         column='download_speed',
                      )])
```
This is discussed in the second example of the documentation (see `corrected_aoi_results` for the column `internet_speed_imputed_mean` vs the `internet_speed_mean` for the aoi tile that is partially overlapped by  
1 data tile.

**Comment by alronlam** [comment]
Thanks butch! Tried it out for mean, and it seems to work. 

Noting there are other cases such as `sum` and `min` where this behavior is desirable too.

For example, take this case of a circular AOI intersecting with one Ookla tile. 
![ookla_data](https://user-images.githubusercontent.com/1049495/185226204-87008eaa-3495-43dd-9d14-3f93f151dc14.png)
![imputed_results](https://user-images.githubusercontent.com/1049495/185226210-8ee2e3b1-cd51-4713-8856-450b1b1e0b0a.png)

* I'd want the number of tests (tests_sum) to just be 2 in this case. But it's becoming 0.29... due to the apportioning logic. However, when I try `imputed_sum` it doesn't yield 2 either (it gives out some large number). 
* I'd also want the `d_mbps_min` to be 3.118, as there is only one intersecting tile. However, it always results to 0. Trying `imputed_min` also doesn't work.
